views/js/previewer/config/item.js

```diff
@@ -0,0 +1,200 @@
+                }
+
+                return parameters;
+            },
+
+            /**
+             * Gets the URI of the service call
+             * @returns {String}
+             */
+            getServiceCallId : function getServiceCallId() {
+                return storage.serviceCallId;
+            },
+
+            /**
+             * Gets the name of the service controller
+             * @returns {String}
+             */
+            getServiceController : function getServiceController() {
+                return storage.bootstrap.serviceController || _defaults.bootstrap.serviceController;
+            },
+
+            /**
+             * Gets the name of the extension containing the service controller
+             * @returns {String}
+             */
+            getServiceExtension : function getServiceExtension() {
+                return storage.bootstrap.serviceExtension || _defaults.bootstrap.serviceExtension;
+            },
+
+            /**
+             * Gets an URL of a service action
+             * @param {String} action - the name of the action to request
+             * @returns {String} - Returns the URL
+             */
+            getTestActionUrl : function getTestActionUrl(action) {
+                return urlUtil.route(action, this.getServiceController(), this.getServiceExtension(), this.getParameters());
+            },
+
+            /**
+             * Gets an URL of a service action related to a particular item
+             * @param {String|Object} itemIdentifier - The URI of the item
+             * @param {String} action - the name of the action to request
+             * @returns {String} - Returns the URL
+             */
+            getItemActionUrl : function getItemActionUrl(itemIdentifier, action) {
+                return urlUtil.route(action, this.getServiceController(), this.getServiceExtension(), this.getParameters(itemIdentifier));
+            },
+
+            /**
+             * Gets an URL of a telemetry signal related to a particular item
+             * @param {String|Object} itemIdentifier - The URI of the item
+             * @param {String} signal - the name of the signal to request
+             * @returns {String} - Returns the URL
+             */
+            getTelemetryUrl : function getTelemetryUrl(itemIdentifier, signal) {
+                return urlUtil.route(signal, this.getServiceController(), this.getServiceExtension(), this.getParameters(itemIdentifier));
+            },
+
+            /**
+             * Gets the request timeout
+             * @returns {Number}
+             */
+            getTimeout : function getTimeout() {
+                return storage.timeout;
+            },
+
+            /**
+             * Gets the config for the communication channel
+             * @returns {Object}
+             */
+            getCommunicationConfig : function getCommunicationConfig() {
```

Comment:
Do we need a communication confifg for the previewer ?

Reply:
Probably not. I kept it for compatibility reasons. I will test if I can get rid of it.